packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1393,13 +1404,11 @@ export default class Tokenizer extends ParserErrors {
           ) {
             if (inTemplate) {
               return null;
-            } else if (this.state.strict) {
-              this.raise(codePos, Errors.StrictNumericEscape);
             } else {
-              // This property is used to throw an error for
-              // an octal literal in a directive that occurs prior
-              // to a "use strict" directive.
-              this.state.octalPositions.push(codePos);
+              this.recordNumericEscapeErrors(
+                codePos,
+                Errors.StrictOctalLiteral,
```

Comment:
We should have thrown StrictNumericEscape since we are parsing an escaped sequence. Will addressed it in another PR.